src/logic/function/basic.lean

```diff
@@ -841,3 +843,10 @@ funext $ λ a, funext $ λ b, (is_symm_op.symm_op a b).symm
 lemma inv_image.equivalence {α : Sort u} {β : Sort v} (r : β → β → Prop) (f : α → β)
   (h : equivalence r) : equivalence (inv_image r f) :=
 ⟨λ _, h.1 _, λ _ _ x, h.2.1 x, inv_image.trans r f h.2.2⟩
+
+instance {α β : Type*} {r : α → β → Prop} [Π a, decidable_pred (r a)] :
+  decidable_pred (uncurry r) :=
+λ x, ‹Π a, decidable_pred (r a)› x.1 x.2
```

Comment:
```suggestion
instance {α β : Type*} {r : α → β → Prop} [decidable_rel r] :
  decidable_pred (uncurry r) :=
λ x, ‹Π a, decidable_pred (r a)› x.1 x.2
```

Reply:
That doesn't work because `r : α → β → Prop`, not `r : α → α → Prop`

Reply:
Whoops, then how about the more symmetric
```suggestion
instance {α β : Type*} {r : α → β → Prop} [Π a b, decidable (r a b)] :
  decidable_pred (uncurry r) :=
λ x, ‹Π a b, decidable (r a b)› x.1 x.2
```
Or perhaps even
```suggestion
instance {α β : Type*} {r : α → β → Prop} (x : α × β) [decidable (r x.1 x.2)] :
  decidable (uncurry r x) :=
‹decidable (r x.1 x.2)›
```

Reply:
Yeah that ends up more principled.